src/main/java/org/apereo/openequella/tools/toolbox/api/OpenEquellaRestUtils.java

```diff
@@ -335,6 +338,67 @@ public boolean gatherAccessToken() {
     return cachedItems;
   }
 
+  public List<ParsedCollection> gatherCollections() throws Exception {
+    CloseableHttpClient httpclient = HttpClients.createDefault();
+    String url = Config.get(Config.OEQ_URL) + "/api/collection";
+    HttpGet http = new HttpGet(url);
+
+    http.addHeader("X-Authorization", "access_token=" + accessToken);
+    LOGGER.debug("Making the API call: {}", url);
+    HttpResponse response = httpclient.execute(http);
+    HttpEntity entity = response.getEntity();
+
+    int statusCode = response.getStatusLine().getStatusCode();
+
+    String respStr = EntityUtils.toString(entity);
+    LOGGER.trace(statusCode);
+    LOGGER.trace(respStr);
+    if (statusCode != 200) {
+      String msg =
+          String.format(
+              "FAILURE accessing openEQUELLA list collection api [%s]:  %s - %s",
+              url, statusCode, respStr);
+      LOGGER.error(msg);
+      throw new Exception(msg);
+    }
+
+    List<ParsedCollection> cache = new ArrayList<>();
+
+    JSONObject searchResults = new JSONObject(respStr);
+    final int available = confirmAndGatherInt(searchResults, "available");
+    final int length = confirmAndGatherInt(searchResults, "length");
+    if (available != length) {
+      String msg =
+          String.format(
+              "FAILURE to retrieve all collections.  Likely need to implement pagination.  Available = [{}], Length = [{}]",
+              available,
+              length);
+      LOGGER.error(msg);
+      throw new Exception(msg);
+    }
+    JSONArray arr = confirmAndGatherJsonArray(searchResults, "results");
+    int resultsLength = arr.length();
+    LOGGER.info(
+        "Requested collections.  API stats: start=[{}] returned length=[{}], available=[{}], # of results=[{}]",
+        cacheStatStart,
+        cacheStatLength,
+        cacheStatAvailable,
+        resultsLength);
+
+    for (int i = 0; i < arr.length(); i++) {
```

Comment:
is the `int resultsLength = arr.length();` needed?
If so could `resultsLength` be reused on line 388?

Reply:
ah good point